File: frontend/src/models/Thread.ts
// ---- state ---- //
export interface Thread {
    threadKey: string;
    title: string;
    contributor: string;
    views: number;
    commentSum: number;
    createDate: string;
    updateDate: string;
}

export interface Threads {
    threads: Array<Thread>;
}

// ---- saga response ---- //
export interface FetchThreadResponse {
    threads: Array<Thread>
}

export interface CreateThreadResponse {
    threadKey: string;
    title: string;
    contributor: string;
    views: number;
    commentSum: number;
    createDate: string;
    updateDate: string;
}

export interface UpdateThreadResponse {
    threadKey: string;
    title: string;
    contributor: string;
    views: number;
    commentSum: number;
    createDate: string;
    updateDate: string;
}

export interface ThreadResponse {
    fetchResponse: {
        pending: boolean;
        success: boolean;
        error: boolean;
        body: FetchThreadResponse;
    };
    createResponse: {
        pending: boolean;
        success: boolean;
        error: boolean;
        body: CreateThreadResponse;
    };
    updateResponse: {
        pending: boolean;
        success: boolean;
        error: boolean;
        body: UpdateThreadResponse;
    };
    deleteResponse: {
        pending: boolean;
        success: boolean;
        error: boolean;
    };
}

File: backend/app/interfaces/response/thread.go
package response

import "github.com/jumpei00/board/backend/app/domain"

type ResponseThreads struct {
	Threads []*ResponseThread `json:"threads"`
}

type ResponseThread struct {
	ThreadKey   string `json:"thread_key"`
	Title       string `json:"title"`
	Contributor string `json:"contributor"`
	Views       int    `json:"views"`
	CommentSum  int    `json:"comment_sum"`
	CreateDate  string `json:"create_date"`
	UpdateDate  string `json:"update_date"`
}

func NewResponseThread(thread *domain.Thread) *ResponseThread {
	return &ResponseThread{
		ThreadKey:   thread.GetKey(),
		Title:       thread.GetTitle(),
		Contributor: thread.GetContributor(),
		Views:       thread.GetViews(),
		CommentSum:  thread.GetCommentSum(),
		CreateDate:  thread.FormatCreatedDate(),
		UpdateDate:  thread.FormatUpdatedDate(),
	}
}
